src/sentry/api/serializers/models/sentry_app.py

```diff
@@ -29,6 +29,9 @@ def serialize(self, obj, attrs, user, access):
             "allowedOrigins": obj.application.get_allowed_origins(),
         }
 
+        if obj.get_status_display() == "published" and obj.date_published:
```

Comment:
@nolachen I think we should use the `SentryAppStatus` class which as the constant for `published`.